src/Symfony/Component/HttpKernel/UriSigner.php

```diff
@@ -48,6 +59,14 @@ public function sign(string $uri): string
             parse_str($url['query'], $params);
         }
 
+        if (isset($params[$this->expirationParameter])) {
+            throw new \LogicException(sprintf('URI query parameter conflict. Parameter name "%s" is reserved.', $this->expirationParameter));
```

Comment:
You should use the LogicExceptiion from HttpKernel component

Reply:
The class moved from **HttpKernel** to **HttpFoundation** and there is no `LogicException `in this component. Can we keep `\LogicException` ?

Reply:
Let's create `LogicException` in the component than.